dask_expr/reductions.py

```diff
@@ -176,7 +176,19 @@ def chunk_kwargs(self):
         return {"ignore_index": self.ignore_index, **self._subset_kwargs()}
 
     def _simplify_up(self, parent):
-        return
+        if self.subset is not None and isinstance(parent, Projection):
```

Comment:
Maybe we don't need the `isinstance(parent, Projection)` condition here.  If we have the same columns then there's no need to project, regardless of the type of the parent operation

Reply:
This might open us up to some edge case where parent isn’t a frame anymore? Saying might, not totally sure

Reply:
Even then I'd hope that we would use `isisntance(parent.columns, pd.Index)` or something similar.  

I think that we shouldn't introduce a Projection if it doesn't help.  I could imagine future optimizations, for example, which remove unnecessary projections, and then we'd be back in a cycle.

Reply:
This doesn't work in all cases unfortunately,

if you do

```
ddf.drop_duplicates(subset="a").index.compute()
```
then parent it an ``Index`` which doesn't have a columns attribute, we could check ``hasattr(parent, "columns")``, but that is kinda ugly as well

Reply:
Exprs don't really have a concept of frame vs series vs index.  They're all the same.  Ideally we can avoid this distinction while in expr-land entirely.

Reply:
So you wouldn't do something like

```
ddf.drop_duplicates(subset="a").index
```
?

That wasn't my idea, but ``assert_eq`` is checking the index and it's meta information as well.

Reply:
> So you wouldn't do something like

You would, but the resulting object would still just be an Expr.  In general we don't know the collection type (DataFrame/Series/Index) of exprs.  We treat them all the same in terms of what API they expose. 

This was one of the original challenges we found in early experiments.  We encountered an explosion of types like `SeriesAdd`, `FrameAdd`, `IndexAdd` which became challenging quickly.  The solution was to have two different class hierarchies, `DataFrame/Series/Index` which understood the different in APi but didn't track any user intent and the `Expr` class hierarchy, which captured what the user wrote down but didn't try to track the `DataFrame/Series/Index` difference at all.  Exprs only track what the user wrote down, and don't try to infer much from it.  Sometimes optimizations may ask `meta`, but even that is a derived quantity rather than something innate about the Expr.

Reply:
Yeah this makes sense (and this is also what I understood so far). But the error I am getting is the following (this all happens on the expression level, but I might be missing something):

If I do 
```
ddf.drop_duplicates(subset="a").index
```

then ``parent`` in ``_simplify_up`` of ``DropDuplicates`` is the expression ``Index``, e.g. ``class Index(Elemwise):``. If I do not check for ``isintance(parent, Projection)``, then I will try to access ``parent.columns``. But the ``columns`` property returns ``self._meta.columns`` and ``self._meta`` is an actual ``pd.Index``, which does not have ``columns``. This is the error I am seeing. So we have to ensure somehow that ``parent`` is an expression where we can access columns in ``_simplify_up``?